x/alliance/keeper/msg_server.go

```diff
@@ -295,14 +299,22 @@ func (m MsgServer) DeleteAlliance(ctx context.Context, msg *types.MsgDeleteAllia
 	if !found {
 		return nil, types.ErrUnknownAsset
 	}
-
-	if asset.TotalTokens.GT(math.ZeroInt()) {
-		return nil, types.ErrActiveDelegationsExists
+	if asset.IsDissolving {
+		return nil, types.ErrAssetDissolving
 	}
 
-	err := m.DeleteAsset(sdkCtx, asset)
-	if err != nil {
-		return nil, err
+	if asset.TotalTokens.GT(math.ZeroInt()) {
+		asset.IsDissolving = true
```

Comment:
WHen deleting an asset, we should make it such that we disable the rewards as well. I'm thinking we make the reward weight to be zero.

Reply:
Hmm.... Good point!